cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -460,7 +460,7 @@ private RestartPlan restartPlan(int podId, Pod pod, RestartContext restartContex
             throw new FatalProblem("Pod is unschedulable");
         }
         // Unless the annotation is present, check the pod is at least ready.
-        boolean needsRestart = reasonToRestartPod != null && !reasonToRestartPod.isEmpty();
+        boolean needsRestart = !reasonToRestartPod.isEmpty();
```

Comment:
Can we stick an `Objects.requireNonNull` around the `List<String> reasonToRestartPod = podNeedsRestart.apply(pod);` a few lines above, so if a function does return null we have a nice obvious NPE?